### ISSUE TITLE
Support for stdout and process arguments in already running processes

### ISSUE THREAD

**Issue by Turnerj** [dotnet/runtime#37851]
The API around the `Process` class seems counter intuitive when handling _already running processes_. For the most part, you can access a variety of properties however if you attempt to access something like `StandardOuput` or `StartInfo`, then you hit an issue.

via `StandardOuput`:
```csharp
var process = Process.GetProcessesByName("existingProcess")[0];
var outputStream = process.StandardOuput; // Throws InvalidOperationException: "StandardOut has not been redirected or the process hasn't started yet."
```

via `StartInfo.Arguments`:
```csharp
var process = Process.GetProcessesByName("existingProcess")[0];
var arguments = process.StartInfo.Arguments; // Throws InvalidOperationException: "Process was not started by this object, so requested information cannot be determined."
```

I don't think this is a bug per-se, I think this was an intentional design choice though I think it would be great to revisit and see whether the decisions that led to it behaving like this are still applicable. Is it due to a permission/security issue between processes or perhaps a limitation in Windows or just not a very common usecase?

**Comment by danmosemsft** [comment]
@Turnerj any interest in hacking a bit to see whether it could usefully be wired up, and inform a discussion?

**Comment by Turnerj** [comment]
Yeah, happy to give it a shot. Haven't done any builds locally of the runtime before so might have a few questions and issues with that. 🙂
